### PR TITLE
extract union_of

### LINKED ISSUE #878: Replace union_of lib with gem
As part of our [week of Open Source](https://x.com/_m27e/status/1823029064372031586), we extracted the internal [`union_of`](https://github.com/keygen-sh/keygen-api/blob/82adaeb9537ab5c4309d29fc4557e26a85683682/lib/union_of.rb) library into an Open Source gem: https://github.com/keygen-sh/union_of. I want to replace the internal lib with the new gem.

Including the `UnionOf::Macro` module is no longer required in models using `union_of:`.

Similar task to https://github.com/keygen-sh/keygen-api/pull/876.

### PR DESCRIPTION
**Pull request by neo773**
/claim #878
/closes #878

### CODE REVIEW

**Comment by ezekg** [comment]
Please be sure to remove any occurrence of [`UnionOf::Macro`](https://github.com/search?q=repo%3Akeygen-sh%2Fkeygen-api%20UnionOf%3A%3AMacro&type=code), as that is no longer required by the gem.

**Comment by neo773** [comment]
@ezekg done

**Comment by ezekg** [comment]
Please squash comments. CI will likely fail since our CI requires secrets and GH doesn't like to allow secrets to third-party PRs.

Will run CI locally regardless. ty 👍

**Comment by ezekg** [comment]
Please also sign the CLA: https://github.com/keygen-sh/keygen-api/pull/884#issuecomment-2347236845

**Comment by neo773** [comment]
> Please also sign the CLA: [#884 (comment)](https://github.com/keygen-sh/keygen-api/pull/884#issuecomment-2347236845)

done

File: Gemfile [hunk 1]
```diff
@@ -59,6 +59,7 @@ gem 'strong_migrations'
 
 # Pattern matching
 gem 'rails-pattern_matching'
+gem 'union_of'
```
Comment (ezekg) [inline Gemfile:62]: This should be under the DB section.

**Pull request merged**